pkg/localstorev2/mock/mockstorer.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mockstorer
```

Comment:
why do we need a mock storer?

Reply:
Usually, for unit tests we mock the objects that the package uses and are not defined in the package. The storer should be tested separately. Having API pkg unit tests depend on the actual storer implementation correctness is not right. There could be bugs in the implementation etc. The goal of the unit tests here is just to ensure the code in the api pkg files is correct.

The type of tests you are mentioning are integration tests. We can have these separately. Or we can develop beekeeper tests which will do the integration tests. We will probably have to change a bunch of these anyway, so we can even add some. Along with this we need to provide debug APIs for showing storer state to verify in the tests.

Reply:
> Having API pkg unit tests depend on the actual storer implementation correctness is not right. There could be bugs in the implementation etc

I don't understand this. We are not mocking here something complex like kademlia that's needs external connections.
We are mocking something that can be created as easily as `storer.New(ctx context.Context, "", opts)`

But I get it, we don't really want to spend the energy rewriting already existing tests.